utils/ort/src/funTest/kotlin/storage/S3FileStorageFunTest.kt

```diff
@@ -0,0 +1,153 @@
+
+        override fun handle(exchange: HttpExchange) {
+            when (exchange.requestMethod) {
+                "HEAD" -> {
+                    val status = if (requests.containsKey(exchange.requestURI.toString())) {
+                        HttpURLConnection.HTTP_OK
+                    } else {
+                        HttpURLConnection.HTTP_NOT_FOUND
+                    }
+                    exchange.sendResponseHeaders(status, -1)
+                }
+                "PUT" -> {
+                    val key = exchange.requestURI.toString().removeSurrounding("/$bucket/", "?x-id=PutObject")
+                    requests[key] = exchange.requestBody.reader().use { it.readText() }
+                    exchange.sendResponseHeaders(HttpURLConnection.HTTP_OK, 0)
+                }
+                "GET" -> {
+                    val key = exchange.requestURI.toString().removeSurrounding("/$bucket/", "?x-id=GetObject")
+                    requests[key].let { data ->
+                        if (data != null) {
+                            exchange.sendResponseHeaders(HttpURLConnection.HTTP_OK, 0)
+                            exchange.responseBody.writer().use { it.write(data) }
+                        } else {
+                            exchange.sendResponseHeaders(HttpURLConnection.HTTP_NOT_FOUND, 0)
+                            exchange.responseBody.bufferedWriter().use {
+                                it.write(
+                                "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" +
+                                    "<Error>\n" +
+                                    "  <Code>NoSuchKey</Code>\n" +
+                                    "  <Message>The resource you requested does not exist</Message>\n" +
+                                    "  <Resource>${exchange.requestURI}</Resource> \n" +
+                                    "  <RequestId>4442587FB7D0A2F9</RequestId>\n" +
+                                    "</Error>"
+                                )
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    // Start a local HTTP server to mock S3 with the system default value for queued incoming connections.
+    private val server = HttpServer.create(InetSocketAddress(loopback, port), 0).apply {
+        createContext("/", handler)
+        start()
+    }
+
+    private val storage = S3FileStorage(
+        bucketName = bucket,
+        awsRegion = "us-east-1",
+        accessKeyId = "key",
+        secretAccessKey = "secret",
+        customEndpoint = "$protocol://${loopback.hostAddress}:$port"
+    )
+
+    override suspend fun afterEach(testCase: TestCase, result: TestResult) {
+        handler.requests.clear()
+    }
+
+    override suspend fun afterSpec(spec: Spec) {
+        // Ensure the server is properly stopped even in case of exceptions, but wait at most 5 seconds.
+        server.stop(5)
+    }
+
+    init {
+
```

Comment:
Please drop this superfluous empty line.

Reply:
Addressed